From b2fb8081edb3f6cc3e73ff0a822cfd4f6ccf9829 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Tue, 22 Nov 2016 22:32:23 +0800
Subject: [PATCH 1288/5242] MLK-13501 ARM: imx: correct i.mx7ulp clock tree
 setting

commit  15def056b6051e5862c1f4a8f1c927aa14de9409 from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX7ULP, there are options to select SPLL or SPLL PFD
as SPLL output clock SPLL_SEL, so the SPLL option for
sys_sel mux should be from SPLL_SEL, NOT from SPLL directly.

Previous:
    spll_pre_sel                          1            1    24000000          0
       spll_pre_div                       1            1    24000000          0
          spll                            2            2   531648000          0
             sys_sel                      1            1   531648000          0
                core_div                  2            2   531648000          0
                   plat_div               1            1   531648000          0
             spll_pfd3                    0            0   979729408          0
             spll_pfd2                    0            0   979729408          0
             spll_pfd1                    0            0   979729408          0
             spll_pfd0                    1            1   503666526          0
                spll_pfd_sel              0            0   503666526          0
                   spll_sel               0            0   503666526          0
After fixed:
    spll_pre_sel                          1            1    24000000          0
       spll_pre_div                       1            1    24000000          0
          spll                            1            1   531648000          0
             spll_pfd3                    0            0   979729408          0
             spll_pfd2                    0            0   979729408          0
             spll_pfd1                    0            0   979729408          0
             spll_pfd0                    2            2   503666526          0
                spll_pfd_sel              1            1   503666526          0
                   spll_sel               1            1   503666526          0
                      sys_sel             1            1   503666526          0
                         core_div           1            1   503666526        0
                            plat_div           1            1   503666526     0

CORE_DIV clock will be enabled automatically when PLAT_DIV
is enabled, so we can skip it in clks_init_on.

Now that sys_sel clock tree is correct, no need to have SPLL_PFD0
in clks_init_on, as it will be enabled automatically because of
PLAT_DIV.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx7ulp.c |    4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/drivers/clk/imx/clk-imx7ulp.c b/drivers/clk/imx/clk-imx7ulp.c
index c612e70..0804bef 100644
--- a/drivers/clk/imx/clk-imx7ulp.c
+++ b/drivers/clk/imx/clk-imx7ulp.c
@@ -24,7 +24,7 @@
 static const char *spll_sels[]		= { "spll", "spll_pfd_sel", };
 static const char *apll_pfd_sels[]	= { "apll_pfd0", "apll_pfd1", "apll_pfd2", "apll_pfd3", };
 static const char *apll_sels[]		= { "apll", "apll_pfd_sel", };
-static const char *sys_sels[]		= { "dummy", "osc", "sirc", "firc", "ckil", "apll", "spll", "upll", };
+static const char *sys_sels[]		= { "dummy", "osc", "sirc", "firc", "ckil", "apll_sel", "spll_sel", "upll", };
 static const char *ddr_sels[]		= { "apll_pfd_sel", "upll", };
 static const char *nic_sels[]		= { "firc", "ddr_div", };
 static const char *periph_plat_sels[]	= { "dummy", "nic1_bus", "nic1_div", "ddr_div", "apll_pfd2", "apll_pfd1", "apll_pfd0", "upll", };
@@ -34,8 +34,6 @@
 static struct clk_onecell_data clk_data;
 
 static int const clks_init_on[] __initconst = {
-	IMX7ULP_CLK_SPLL_PFD0,
-	IMX7ULP_CLK_CORE_DIV,
 	IMX7ULP_CLK_BUS_DIV,
 	IMX7ULP_CLK_PLAT_DIV,
 	IMX7ULP_CLK_NIC0_DIV,
-- 
1.7.9.5

